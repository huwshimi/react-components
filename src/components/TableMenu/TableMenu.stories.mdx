import { ArgsTable, Canvas, Meta, Story } from "@storybook/addon-docs/blocks";

import TableMenu from "./TableMenu";

<Meta
  title="TableMenu"
  component={TableMenu}
  argTypes={{
    children: {
      control: {
        type: "text",
      },
    },
  }}
/>

export const Template = (args) => <TableMenu {...args} />;

### TableMenu

This is a double content row for use in tables.

### Props

<ArgsTable of={TableMenu} />

### TableMenu

<Canvas>
  <Story
    name="Default"
    args={{ primary: "eth0", secondary: "0d:be:38:1d:1c:a9" }}
  >
    {Template.bind({})}
  </Story>
</Canvas>
