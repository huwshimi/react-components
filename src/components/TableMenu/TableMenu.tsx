import classNames from "classnames";

import ContextualMenu from "../ContextualMenu";
import type { Props as ContextualMenuProps } from "../ContextualMenu/ContextualMenu";

export type Props<L = null> = {
  className?: ContextualMenuProps<L>["className"] | null;
  disabled?: ContextualMenuProps<L>["toggleDisabled"] | null;
  links?: ContextualMenuProps<L>["links"] | null;
  onToggleMenu?: ContextualMenuProps<L>["onToggleMenu"] | null;
  position?: ContextualMenuProps<L>["position"] | null;
  positionNode?: ContextualMenuProps<L>["positionNode"] | null;
  title?: string | null;
};

const TableMenu = <L extends null>({
  className,
  disabled = false,
  links,
  title,
  onToggleMenu,
  position = "left",
  positionNode,
}: Props<L>): JSX.Element => {
  // If there are no links then make it an empty array so that it can be validly spread below.
  links = links || [];
  return (
    <ContextualMenu
      className={classNames("p-table-menu", className)}
      hasToggleIcon
      links={[
        ...(title ? [title] : []),
        ...(Array.isArray(links) ? links : [links]),
      ]}
      onToggleMenu={onToggleMenu}
      position={position}
      positionNode={positionNode}
      toggleAppearance="base"
      toggleClassName="u-no-margin--bottom p-table-menu__toggle"
      toggleDisabled={disabled}
    />
  );
};

export default TableMenu;
