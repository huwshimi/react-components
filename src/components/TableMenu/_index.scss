@mixin TableMenu {
  $icon-size: map-get($icon-sizes, default);
  $icon-space: $icon-size + $sph-inner--small + $sph-inner;

  .p-table-menu--hasIcon .p-contextual-menu__non-interactive {
    padding-left: $icon-space;
  }

  .p-table-menu .p-contextual-menu__dropdown {
    margin-left: -$sph-inner;
    min-width: 13rem;

    .p-table-menu__icon-space,
    [class*="p-icon"] {
      margin-right: $sph-inner--small;
    }
  }

  .p-table-menu--hasIcon .p-contextual-menu__dropdown {
    margin-left: -$icon-space;
  }

  .p-table-menu__toggle {
    line-height: $icon-size;
    padding: 0 $icon-size / 2;
  }

  .p-table-menu__icon-space {
    $icon-space: map-get($icon-sizes, default);
    display: inline-block;
    width: $icon-space;
  }
}
