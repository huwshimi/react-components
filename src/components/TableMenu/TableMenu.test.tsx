import { shallow } from "enzyme";

import TableMenu from "./TableMenu";

describe("TableMenu ", () => {
  it("renders", () => {
    const wrapper = shallow(
      <TableMenu links={[{ children: "Item1" }]} title="Actions:" />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
