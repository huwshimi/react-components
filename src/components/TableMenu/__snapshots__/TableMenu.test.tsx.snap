// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableMenu  renders 1`] = `
<ContextualMenu
  className="p-table-menu"
  hasToggleIcon={true}
  links={
    Array [
      "Actions:",
      Object {
        "children": "Item1",
      },
    ]
  }
  position="left"
  toggleAppearance="base"
  toggleClassName="u-no-margin--bottom p-table-menu__toggle"
  toggleDisabled={false}
/>
`;
